DirectX11Proj/d3dSwapchain.cpp

```diff
@@ -0,0 +1,152 @@
+#include "d3dSwapchain.h"
+
+#include <minwinbase.h>
+#include <cassert>
+#include <iostream>
+
+
+d3dSwapchain::d3dSwapchain(IDXGIFactory* const aFactory, ID3D11Device* const aDevice) : mPFactory(aFactory), mDevice(aDevice)
+{
+	assert(aFactory != nullptr);
+	assert(aDevice != nullptr);
+}
+
+
+d3dSwapchain::~d3dSwapchain()
+{
+}
+
+void d3dSwapchain::Shutdown()
+{
+	if (mSwapChain)
+	{
+		mSwapChain->SetFullscreenState(false, NULL);
```

Comment:
idk if it asks for a pointer, but if it does return a nullptr instead (easier for debugging purposes)